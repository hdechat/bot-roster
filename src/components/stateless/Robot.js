import React from 'react';

function Robot({ robot }) {
  const { 
    id, firstName, lastName, speed, 
    strength, agility, totalAttrScore } = robot;

  return (
    <tr>
      <td>{id}</td>
      <td>{firstName}</td>
      <td>{lastName}</td>
      <td>{speed}</td>
      <td>{strength}</td>
      <td>{agility}</td>
      <td>{totalAttrScore}</td>
    </tr>
  );
};

export default Robot;
