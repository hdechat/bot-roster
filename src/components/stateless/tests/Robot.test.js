import React from 'react';
import Robot from '../Robot';
import { shallow, mount } from 'enzyme';

describe('Robot', () => {
  let wrapper;
  let mockRobot;
  let mockUpdateName;
  let mockDeleteRobot;
  let mockEventChangeFirstName;
  let mockEventChangeLastName;
  let mockEventChangeLastNameWithTab;

  beforeEach(() => {
    mockEventChangeFirstName = {
      target: {
        classList: ['firstName'],
        innerText: 'Buck'
      },
      preventDefault: jest.fn(),
      which: 13
    };

    mockEventChangeLastName = {
      target: {
        classList: ['lastName'],
        innerText: 'Bot'
      },
      preventDefault: jest.fn(),
      which: 13
    }

    mockEventChangeLastNameWithTab = {
      target: {
        classList: ['lastName'],
        innerText: 'Bot'
      },
      preventDefault: jest.fn(),
      which: 9
    }

    mockRobot = {
      id: 'ABC123',
      firstName: 'Twikki',
      lastName: 'Rogers',
      speed: 20,
      strength: 30,
      agility: 40,
      totalAttrScore: 90
    };

    mockUpdateName = jest.fn();
    mockDeleteRobot = jest.fn();

    wrapper = shallow(<Robot 
      team={''}
      robot={mockRobot} 
      updateName={mockUpdateName}
      deleteRobot={mockDeleteRobot}
      inRosters={false}/>);
  });

  it('matches snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('calls updateName with the correct params if user edits first name then presses enter', () => {
    wrapper.find('.firstName').simulate('keypress', mockEventChangeFirstName);

    const expected = {...mockRobot, firstName: 'Buck'}

    expect(mockUpdateName).toHaveBeenCalledWith(expected);
  });

  it('calls updateName with the correct params if user edits first name, leaves element, and presses Enter', () => {
    wrapper.find('.firstName').prop('onBlur')(mockEventChangeFirstName)

    const expected = {...mockRobot, firstName: 'Buck'}

    expect(mockUpdateName).toHaveBeenCalledWith(expected);
  });

  it('calls updateName with the correct params if user edits last name then presses enter', () => {
    wrapper.find('.lastName').simulate('keypress', mockEventChangeLastName);

    const expected = {...mockRobot, lastName: 'Bot'}

    expect(mockUpdateName).toHaveBeenCalledWith(expected);
  });

  it('calls updateName with the correct params if user edits last name, leaves element then presses enter', () => {
    wrapper.find('.lastName').prop('onBlur')(mockEventChangeLastName)

    const expected = {...mockRobot, lastName: 'Bot'}

    expect(mockUpdateName).toHaveBeenCalledWith(expected);
  });

  it('calls updateName with the correct params if user edits last name then presses tab', () => {
    wrapper.find('.lastName').simulate('keypress', mockEventChangeLastNameWithTab);

    const expected = {...mockRobot, lastName: 'Bot'}

    expect(mockUpdateName).toHaveBeenCalledWith(expected);
  });

  it('calls deleteRobot with the correct params on button click', () => {
    wrapper.find('button').simulate('click')

    expect(mockDeleteRobot).toHaveBeenCalledWith(mockRobot);
  });
  
  it('does not render delete button if at rosters endpoint', () => {
    wrapper = shallow(<Robot 
      team={''}
      robot={mockRobot} 
      updateName={mockUpdateName}
      deleteRobot={mockDeleteRobot}
      inRosters={true}/>)

    expect(wrapper.find('button').length).toEqual(0)
  });
});
