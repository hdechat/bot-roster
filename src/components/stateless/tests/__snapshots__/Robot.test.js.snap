// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Robot matches snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Robot
    robot={
        Object {
            "agility": 40,
            "firstName": "Twikki",
            "id": "ABC123",
            "lastName": "Rogers",
            "speed": 20,
            "strength": 30,
            "totalAttrScore": 90,
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <td>
          ABC123
</td>,
        <td>
          Twikki
</td>,
        <td>
          Rogers
</td>,
        <td>
          20
</td>,
        <td>
          30
</td>,
        <td>
          40
</td>,
        <td>
          90
</td>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "ABC123",
        },
        "ref": null,
        "rendered": "ABC123",
        "type": "td",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Twikki",
        },
        "ref": null,
        "rendered": "Twikki",
        "type": "td",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Rogers",
        },
        "ref": null,
        "rendered": "Rogers",
        "type": "td",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": 20,
        },
        "ref": null,
        "rendered": 20,
        "type": "td",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": 30,
        },
        "ref": null,
        "rendered": 30,
        "type": "td",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": 40,
        },
        "ref": null,
        "rendered": 40,
        "type": "td",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": 90,
        },
        "ref": null,
        "rendered": 90,
        "type": "td",
      },
    ],
    "type": "tr",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <td>
            ABC123
</td>,
          <td>
            Twikki
</td>,
          <td>
            Rogers
</td>,
          <td>
            20
</td>,
          <td>
            30
</td>,
          <td>
            40
</td>,
          <td>
            90
</td>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "ABC123",
          },
          "ref": null,
          "rendered": "ABC123",
          "type": "td",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Twikki",
          },
          "ref": null,
          "rendered": "Twikki",
          "type": "td",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Rogers",
          },
          "ref": null,
          "rendered": "Rogers",
          "type": "td",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": 20,
          },
          "ref": null,
          "rendered": 20,
          "type": "td",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": 30,
          },
          "ref": null,
          "rendered": 30,
          "type": "td",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": 40,
          },
          "ref": null,
          "rendered": 40,
          "type": "td",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": 90,
          },
          "ref": null,
          "rendered": 90,
          "type": "td",
        },
      ],
      "type": "tr",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getSnapshotBeforeUpdate": true,
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
